### DISCUSSION TITLE
Multiplying transformation matrix with multiple vectors.

### DISCUSSION THREAD

**Discussion by 5nefarious** [bitshifter/glam-rs#155] [Q&A] [answered]
Is there a way to put a bunch of `Vec3`s in an 3xN matrix, so that I can transform all of them with a single matrix multiplication?

**Comment by bitshifter** [comment]
There isn't currently, but it would be a good thing to have. The main question is what to do with the results. The easy option would be to mutate the input but it might also be desirable to support a separate output.

**Reply by bitshifter** [reply, marked as answer]
I guess this could be handled by Rust iterators, e.g. something like

```
vec3s.iter().map(|v| m.transform_point3(v)).collect()
```

or to mutate the input

```
vec3s.iter_mut().for_each(|v| *v = m.transform_point3(*v));
```
